pylint/checkers/format.py

```diff
@@ -563,15 +558,14 @@ def _check_multi_statement_line(self, node: nodes.NodeNG, line: int) -> None:
         ):
             return
 
-        # Function overloads that use ``Ellipsis`` are exempted.
+        # Functions stubs with ``Ellipsis`` as body are exempted.
         if (
-            isinstance(node, nodes.Expr)
+            isinstance(node.parent, (nodes.FunctionDef, nodes.AsyncFunctionDef))
```

Comment:
```suggestion
            isinstance(node.parent, nodes.FunctionDef)
```

`AsyncFunctionDef` is a [subclass](https://github.com/PyCQA/astroid/blob/main/astroid/nodes/scoped_nodes/scoped_nodes.py#L1778) of `FunctionDef`.

Reply:
Ah, alright, thanks!